src/App.tsx

```diff
@@ -15,36 +16,115 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetically,
+  Length,
+  Reset,
+}
+
+export function sortGoods(
+  goods: string[],
+  type: SortType,
+  reverse: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  switch (type) {
+    case SortType.Alphabetically:
+      preparedGoods.sort((goodA, goodB) => goodA.localeCompare(goodB));
+      break;
+    case SortType.Length:
+      preparedGoods.sort((goodA, goodB) => goodA.length - goodB.length);
+      break;
+    default:
+      break;
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>(goodsFromServer);
+  const [sort, setSort] = useState<SortType | null>(null);
+  const [reverse, setReverse] = useState<boolean>(false);
+
+  const handleSort = (type: SortType) => {
+    const sortedGoods = sortGoods(goodsFromServer, type, reverse);
+
+    setSort(type);
+    setGoods(sortedGoods);
+  };
```

Comment:
The function name `handleSort` should start with a verb that indicates what the function does. A better name might be `handleSortClick` to clarify that it's handling a click event.